src/PlaywrightSharp.Chromium/ChromiumPage.cs

```diff
@@ -251,6 +287,94 @@ private void Client_MessageReceived(object sender, IChromiumEvent e)
+                //
+                // Ignore these messages since:
+                // - there's no execution context we can use to operate with message
+                //   arguments
+                // - these messages are reported before Playwright clients can subscribe
+                //   to the 'console'
+                //   page event.
+                //
+                // @see https://github.com/GoogleChrome/puppeteer/issues/3865
+                return;
+            }
+
+            var context = _contextIdToContext[e.ExecutionContextId.Value];
+            var values = e.Args.Select(arg => context.CreateHandle(arg));
+            Page.AddConsoleMessage(e.Type.ToEnum<ConsoleType>(), values.ToArray(), ToConsoleMessageLocation(e.StackTrace));
+        }
+
+        private async Task OnAttachedToTargetAsync(TargetAttachedToTargetChromiumEvent e)
+        {
+            if (e.TargetInfo.Type != "worker")
+            {
+                return;
+            }
+
+            string url = e.TargetInfo.Url;
+            var session = ChromiumConnection.FromSession(Client).GetSession(e.SessionId);
+            var worker = new Worker(url);
+            Page.AddWorker(e.SessionId, worker);
+
+            void HandleRuntimeExecutionContextCreated(object sender, IChromiumEvent e)
+            {
+                switch (e)
+                {
+                    case RuntimeExecutionContextCreatedChromiumEvent runtimeExecutionContextCreated:
+
+                        worker.CreateExecutionContext(new ChromiumExecutionContext(session, runtimeExecutionContextCreated.Context));
+                        session.MessageReceived -= HandleRuntimeExecutionContextCreated;
+                        break;
+
+                    case RuntimeConsoleAPICalledChromiumEvent runtimeConsoleAPICalled:
+                        var args = runtimeConsoleAPICalled.Args.Select(o =>
+                            worker.ExistingExecutionContext.CreateHandle(o));
+                        Page.AddConsoleMessage(
+                            runtimeConsoleAPICalled.Type.ToEnum<ConsoleType>(),
+                            args.ToArray(),
+                            ToConsoleMessageLocation(runtimeConsoleAPICalled.StackTrace));
+                        break;
+
+                    case RuntimeExceptionThrownChromiumEvent runtimeExceptionThrown:
+                        Page.OnPageError(ExceptionToError(runtimeExceptionThrown.ExceptionDetails));
+                        break;
+                }
+            }
+
+            session.MessageReceived += HandleRuntimeExecutionContextCreated;
+
+            try
+            {
+                await Task.WhenAll(
+                    session.SendAsync(new RuntimeEnableRequest()),
+                    session.SendAsync(new NetworkEnableRequest())).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.WriteLine(ex);

```

Comment:
shouldn't we throw or something?

Reply:
We should translate `.catch(debugError);` to error logging, but not throwing.

Reply:
and in the future `logger.LogDebug` ;)